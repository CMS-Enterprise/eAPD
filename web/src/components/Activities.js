import PropTypes from 'prop-types';
import React from 'react';

import ActivityDetails from './ActivityDetails';
import ActivityList from './ActivityList';
import Section from './Section';
import SectionTitle from './SectionTitle';

const Activities = props => (
  <Section>
    <SectionTitle>Program Activities</SectionTitle>
    <ActivityList {...props} />
    <ActivityDetails {...props} />
  </Section>
);

Activities.propTypes = {
  activities: PropTypes.array.isRequired,
  addActivity: PropTypes.func.isRequired,
  editActivityChecks: PropTypes.func.isRequired,
  editActivityText: PropTypes.func.isRequired
};

export default Activities;
