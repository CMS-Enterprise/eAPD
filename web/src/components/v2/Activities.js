import PropTypes from 'prop-types';
import React from 'react';

import ActivityDescription from './ActivityDescription';
import ContractorResources from './ContractorResources';
import CostAllocation from './CostAllocation';
import Expenses from './Expenses';
import Section from './Section';
import SectionTitle from './SectionTitle';
import StatePersonnel from './StatePersonnel';
import Collapsible from '../Collapsible';
import FormActivityGoals from '../FormActivityGoals';
import FormActivitySchedule from '../FormActivitySchedule';
import { STANDARDS, activityDisplay } from '../../util';

const Activities = ({
  activities,
  addActivity,
  editActivityChecks,
  editActivityText
}) => (
  <Section>
    <SectionTitle>Program Activities</SectionTitle>
    <Collapsible title="Activity List">
      <div className="mb2">
        {activities.map((a, i) => (
          <div key={a.id} className="flex items-center mb1">
            <div className="mr1 bold mono">{i + 1}.</div>
            <div className="mr1 col-4">
              <input
                type="text"
                className="col-12 input m0"
                name={`${a.id}.name`}
                value={a.name}
                onChange={editActivityText}
              />
            </div>
            <div>
              {['HIT', 'HIE', 'MMIS'].map(aType => (
                <label key={aType} className="mr1">
                  <input
                    type="checkbox"
                    name={`${a.id}.type`}
                    value={aType}
                    checked={a.type.includes(aType)}
                    onChange={editActivityChecks}
                  />
                  {aType}
                </label>
              ))}
            </div>
          </div>
        ))}
      </div>
      <button
        type="button"
        className="mt2 btn btn-primary"
        onClick={addActivity}
      >
        Add activity
      </button>
    </Collapsible>

    {activities.map((a, i) => (
      <Collapsible
        key={a.id}
        title={`Program Activities › ${activityDisplay(a, i + 1)}`}
        bgColor="darken-1"
      >
        <Collapsible title="Activity Description">
          <ActivityDescription activity={a} />
        </Collapsible>

        <Collapsible title="Needs and Objectives">
          <FormActivityGoals form={`activity-${a.id}-goals`} />
        </Collapsible>

        <Collapsible title="Proposed Activity Schedule">
          <FormActivitySchedule form={`activity-${a.id}-schedule`} />
        </Collapsible>

        <Collapsible title="State Personnel">
          <StatePersonnel activity={a} />
        </Collapsible>

        <Collapsible title="Contractor Resources">
          <ContractorResources activity={a} />
        </Collapsible>

        <Collapsible title="Expenses">
          <Expenses activity={a} />
        </Collapsible>

        <Collapsible title="Cost Allocation and Other Funding Sources">
          <CostAllocation />
        </Collapsible>

        <Collapsible title="Standards & Conditions">
          <p>Tell us how you’ll meet the Medicaid standards and conditions.</p>
          {STANDARDS.map(std => {
            const inputId = `a-${a.id}-standards-${std.id}`;
            return (
              <div key={std.id}>
                <h3>{std.title}</h3>
                <p>
                  Sed ut perspiciatis unde omnis iste natus error sit voluptatem
                  accusantium doloremque laudantium, totam rem aperiam, eaque
                  ipsa quae ab illo inventore veritatis et quasi architecto...
                </p>
                <div>
                  <label htmlFor={inputId}>
                    Describe how you’ll meet this condition
                  </label>
                  <textarea
                    className="m0 textarea md-col-8"
                    id={inputId}
                    rows="3"
                    spellCheck="true"
                  />
                </div>
              </div>
            );
          })}
        </Collapsible>
      </Collapsible>
    ))}
  </Section>
);

Activities.propTypes = {
  activities: PropTypes.array.isRequired,
  addActivity: PropTypes.func.isRequired,
  editActivityChecks: PropTypes.func.isRequired,
  editActivityText: PropTypes.func.isRequired
};

export default Activities;
