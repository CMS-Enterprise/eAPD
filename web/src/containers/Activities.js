import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { t } from '../i18n';

import ActivityDetailAll from './ActivityDetailAll';
import ActivityListEntry from './ActivityListEntry';
import { addActivity as addActivityAction } from '../actions/activities';
import { Section, Subsection } from '../components/Section';

const Activities = ({ activityIds, addActivity }) => (
  <Section id="activities" resource="activities">
    <Subsection resource="activities.list" open>
      {activityIds.length === 0 ? (
        <div className="mb2 p1 h6 alert">
          {t('activities.noActivityMessage')}
        </div>
      ) : (
        <div className="mb3">
          {activityIds.map((aId, idx) => (
            <ActivityListEntry key={aId} aId={aId} num={idx + 1} />
          ))}
        </div>
      )}
      <button type="button" className="btn btn-primary" onClick={addActivity}>
        {t('activities.addActivityButtonText')}
      </button>
    </Subsection>
    {activityIds.map((aId, idx) => (
      <ActivityDetailAll key={aId} aId={aId} num={idx + 1} />
    ))}
  </Section>
);

Activities.propTypes = {
  activityIds: PropTypes.array.isRequired,
  addActivity: PropTypes.func.isRequired
};

const mapStateToProps = ({ activities }) => ({
  activityIds: activities.allIds
});

const mapDispatchToProps = {
  addActivity: addActivityAction
};

export default connect(mapStateToProps, mapDispatchToProps)(Activities);
