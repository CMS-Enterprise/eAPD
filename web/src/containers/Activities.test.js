import { shallow } from 'enzyme';
import sinon from 'sinon';
import React from 'react';

import { raw as Activities, mapStateToProps } from './Activities';

describe('the Activities list component', () => {
  const props = {
    activityIds: ['1', '2', '3'],
    addActivity: sinon.stub()
  };

  test('renders correctly', () => {
    const component = shallow(<Activities {...props} />);
    expect(component).toMatchSnapshot();
  });

  test('adds a new activity', () => {
    const component = shallow(<Activities {...props} />);
    component.find('button').simulate('click');
    expect(props.addActivity.callCount).toBe(1);
  });

  test('maps redux state to component props', () => {
    expect(
      mapStateToProps({ activities: { allIds: ['1', 'two', '3'] } })
    ).toEqual({ activityIds: ['1', 'two', '3'] });
  });
});
