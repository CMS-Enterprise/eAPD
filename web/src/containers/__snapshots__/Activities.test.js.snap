// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Root component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Activities
    activityIds={
      Array [
        "1",
        "2",
        "3",
      ]
    }
    addActivity={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Subsection
          isKey={false}
          open={true}
          resource="activities.list"
        >
          <div
            className="mb3"
          >
            <Connect(ActivityListEntry)
              aId="1"
              num={1}
            />
            <Connect(ActivityListEntry)
              aId="2"
              num={2}
            />
            <Connect(ActivityListEntry)
              aId="3"
              num={3}
            />
          </div>
          <button
            className="btn btn-primary"
            onClick={[Function]}
            type="button"
          >
            Add activity
          </button>
        </Subsection>,
        Array [
          <Connect(ActivityDetailAll)
            aId="1"
            num={1}
          />,
          <Connect(ActivityDetailAll)
            aId="2"
            num={2}
          />,
          <Connect(ActivityDetailAll)
            aId="3"
            num={3}
          />,
        ],
      ],
      "id": "activities",
      "resource": "activities",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="mb3"
            >
              <Connect(ActivityListEntry)
                aId="1"
                num={1}
              />
              <Connect(ActivityListEntry)
                aId="2"
                num={2}
              />
              <Connect(ActivityListEntry)
                aId="3"
                num={3}
              />
            </div>,
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              Add activity
            </button>,
          ],
          "isKey": false,
          "open": true,
          "resource": "activities.list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(ActivityListEntry)
                  aId="1"
                  num={1}
                />,
                <Connect(ActivityListEntry)
                  aId="2"
                  num={2}
                />,
                <Connect(ActivityListEntry)
                  aId="3"
                  num={3}
                />,
              ],
              "className": "mb3",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "aId": "1",
                  "num": 1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "class",
                "props": Object {
                  "aId": "2",
                  "num": 2,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "class",
                "props": Object {
                  "aId": "3",
                  "num": 3,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add activity",
              "className": "btn btn-primary",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Add activity",
            "type": "button",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "aId": "1",
          "num": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "aId": "2",
          "num": 2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "aId": "3",
          "num": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Subsection
            isKey={false}
            open={true}
            resource="activities.list"
          >
            <div
              className="mb3"
            >
              <Connect(ActivityListEntry)
                aId="1"
                num={1}
              />
              <Connect(ActivityListEntry)
                aId="2"
                num={2}
              />
              <Connect(ActivityListEntry)
                aId="3"
                num={3}
              />
            </div>
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              Add activity
            </button>
          </Subsection>,
          Array [
            <Connect(ActivityDetailAll)
              aId="1"
              num={1}
            />,
            <Connect(ActivityDetailAll)
              aId="2"
              num={2}
            />,
            <Connect(ActivityDetailAll)
              aId="3"
              num={3}
            />,
          ],
        ],
        "id": "activities",
        "resource": "activities",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div
                className="mb3"
              >
                <Connect(ActivityListEntry)
                  aId="1"
                  num={1}
                />
                <Connect(ActivityListEntry)
                  aId="2"
                  num={2}
                />
                <Connect(ActivityListEntry)
                  aId="3"
                  num={3}
                />
              </div>,
              <button
                className="btn btn-primary"
                onClick={[Function]}
                type="button"
              >
                Add activity
              </button>,
            ],
            "isKey": false,
            "open": true,
            "resource": "activities.list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(ActivityListEntry)
                    aId="1"
                    num={1}
                  />,
                  <Connect(ActivityListEntry)
                    aId="2"
                    num={2}
                  />,
                  <Connect(ActivityListEntry)
                    aId="3"
                    num={3}
                  />,
                ],
                "className": "mb3",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "class",
                  "props": Object {
                    "aId": "1",
                    "num": 1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "class",
                  "props": Object {
                    "aId": "2",
                    "num": 2,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "class",
                  "props": Object {
                    "aId": "3",
                    "num": 3,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add activity",
                "className": "btn btn-primary",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Add activity",
              "type": "button",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "aId": "1",
            "num": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "aId": "2",
            "num": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "aId": "3",
            "num": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
